packages/tools/devtools/devtools-core/src/data-visualization/DataEditing.ts

```diff
@@ -5,39 +5,37 @@
 
 import { ISharedObject } from "@fluidframework/shared-object-base";
 import { Serializable } from "@fluidframework/datastore-definitions";
-import { EditType } from "../CommonInterfaces";
-
-// Ideas:
-// - Hold onto previous summary and only transmit diff?
-
-// TODOs:
-// - Dependency tracking
-//   - When a particular DDS is no longer reachable via the input data, we need to remove it from the map and stop
-//     emitting updates.
+import { EditType, FluidObjectId } from "../CommonInterfaces";
 
 /**
- * The type of a shared object.
+ * Generates a description of the edit to be applied to {@link @fluidframework/shared-object-base#ISharedObject}'s
+ * current state.
  *
- * @remarks
+ * @param sharedObject - The {@link ISharedObject} whose data will be edited.
+ * @param edit - Describes what changes will be made using {@link Edit}.
+ * @returns - Nothing.
  *
- * This can be acquired via {@link @fluidframework/datastore-definitions#IChannelFactory.Type} field of
- * your shared object's factory class.
+ * @public
  */
-export type SharedObjectType = string;
+export type EditSharedObject = (sharedObject: ISharedObject, edit: Edit) => Promise<void>;
 
 /**
- * Generates a visual description of the provided {@link @fluidframework/shared-object-base#ISharedObject}'s
- * current state.
- *
- * @param sharedObject - The object whose data will be rendered.
- * @param visualizeChildData - Callback to render child content of the shared object.
- *
- * @returns A visual tree representation of the provided `sharedObject`.
- *
+ * Interface to contain information necesary for an edit
  * @public
  */
-export type EditSharedObject = (
-	sharedObject: ISharedObject,
-	data: Serializable,
-	type: EditType,
-) => Promise<void>;
+export interface Edit {
```

Comment:
I know we discussed leaving this type alone, but looking a bit closer at `EditSharedObject`, I think I might actually be leaning towards separate types for edits with the ID and without. We could probably do something like `SharedObjectEdit` for the version that includes the ID. E.g.:

```typescript
export interface Edit {
	/**
	 * Type contains the {@link EditType} of the edit being preformed
	 */
	type: EditType;

	/**
	 * Data contains the new data that will be edited into the DDS
	 */
	data: Serializable<unknown>;
}

export interface SharedObjectEdit extends Edit, HasFluidObjectId { }
```

Reply:
But this isn't critical. Feel free to leave this alone if you would prefer :)

Reply:
This involves adding HasFluidObjectID package as a dependency, thats one of the reasons I didn't do it out of the gate. Should I add the deendency?

Reply:
That interface lives in another module within this same package. Definitely feel free to use it across the devtools stuff. No worries about that.